prow/plugins/label/label_test.go

```diff
@@ -389,19 +507,44 @@ func TestLabel(t *testing.T) {
 		},
 	}
 
+	fakeRepoFunctions := []func(string, string, []string, []string) (*fakegithub.FakeClient, assignEvent){
+		getFakeRepoIssueComment,
+		getFakeRepoIssue,
+		getFakeRepoPullRequest,
+	}
+
 	for _, tc := range testcases {
-		fakeClient, ice := getFakeRepo(tc.body, tc.commenter, tc.repoLabels, tc.issueLabels)
-		if err := handle(fakeClient, logrus.WithField("plugin", pluginName), ice); err != nil {
-			t.Errorf("For case %s, didn't expect error from label test: %v", tc.name, err)
-			continue
-		}
+		for i := 0; i < len(fakeRepoFunctions); i++ {
+			fakeClient, ae := fakeRepoFunctions[i](tc.body, tc.commenter, tc.repoLabels, tc.issueLabels)
 
-		if len(tc.expectedNewLabels) > 0 && !reflect.DeepEqual(tc.expectedNewLabels, fakeClient.LabelsAdded) {
-			t.Errorf("For test %v,\n\tExpected Added %+v \n\tFound %+v", tc.name, tc.expectedNewLabels, fakeClient.LabelsAdded)
-		}
+			if err := handle(fakeClient, logrus.WithField("plugin", pluginName), ae); err != nil {
+				t.Errorf("For case %s, didn't expect error from label test: %v", tc.name, err)
+				return
+			}
+
+			if len(tc.expectedNewLabels) != len(fakeClient.LabelsAdded) {
+				t.Errorf("For test %v,\n\tExpected %+v \n\tFound %+v", tc.name, tc.expectedNewLabels, fakeClient.LabelsAdded)
+				return
+			}
+
+			for i := range tc.expectedNewLabels {
+				if tc.expectedNewLabels[i] != fakeClient.LabelsAdded[i] {
```

Comment:
What if it adds them in a different order than your test expects? It shouldn't matter.